sdk/identity/azure-identity/src/client_secret_credential.cpp

```diff
@@ -9,6 +9,40 @@
 #include <chrono>
 #include <sstream>
 
+namespace {
+// Assumes !scopes.empty()
+std::string FormatScopes(std::vector<std::string> const& scopes, bool asResource)
+{
+  if (asResource && scopes.size() == 1)
+  {
+    auto resource = scopes[0];
+    constexpr char suffix[] = "/.default";
+    constexpr int suffixLen = sizeof(suffix) - 1;
```

Comment:
Why are we doing `-1`. Add a comment.

Reply:
I don't think this necessarily requires a comment, this is the part of the language - when an array is initialized from a literal, `\0` is added to the array - https://en.cppreference.com/w/c/language/array_initialization.

Reply:
> I don't think this necessarily requires a comment

Source comments are rarely **necessary**. It's just meant for helpful hints for future maintainability when code changes, moves around. What happens if I change line 19 tomorrow to `std::string`, now line 20 is broken, if I don't update `-1`, correct?

It's up to your discretion, as the PR author, what's clear/obvious by just reading the code.

Reply:
I can add it, but if we change it to `std::string`, `suffixLen` should be removed entirely, it is no longer necessary. So, the only context in which it makes sense, is when `suffix` is a C-string, so it is 0-terminated, so that "- 1" is needed.

But even without the "-1", `sizeof(std::string())` is broken. It will be the size of the `std::string` class, not the length of the string. `length()` should be used.